### ISSUE TITLE
Reorganize tables that hold label and panorama metadata

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#2741]
#### Brief description of problem/feature
**UPDATE**: After some more investigation and discussion, the final list of things to change can be found in [this comment](https://github.com/ProjectSidewalk/SidewalkWebpage/issues/2741#issuecomment-1130472714). Reasoning for how we get there can be seen by reading this whole thread.

Right now our database has a `label` table. Separate from that are the tables `label_severity`, `label_temporariness`, and `label_description`. These tables could easily be converted into nullable columns of the `label` table. This would simplify writing queries since we would no longer have to do outer joins with these tables to get these critical pieces of information about a label!

I could be convinced that the label description does not need to be included in the `label` table and could remain as it's own table since that is the user-provided description and is rarely used.

I am happy with the label tags having their own table, since there is a one to many relationship there.

There is also a table called `label_point` which holds a lot of the metadata about a label that is used to place it back onto an image. Columns include `canvas_x`, `canvas_y`, `geom`, `heading`, `pitch`, etc. I kind of like the separation between the main metadata for a label being in the `label` table, with the metadata relating to the _location_ of the label being in the `label_point` table.

There is also a table called `gsv_data`, which holds information about the panoramas. I like this. My only problem is that there are a few columns in the `label` table that seem like they should really be in the `gsv_data` table. Namely: `panorama_lat`, `panorama_lng`, `photographer_heading`, and `photographer_pitch`. I believe that those are specific to a panorama, not the location of a label within the panorama, so those should be moved to the appropriate place.

Some benefits of these changes:
1. Our database schema will just make more sense. Better for me. Better for new interns that are being onboarded.
2. We won't have to do so many outer joins to get critical pieces of information about a label, like the severity. This is particularly impactful given the older version of the Slick library that we're using that doesn't deal with outer joins well.

**IN SUMMARY**
1. Move `panorama_lat`, `panorama_lng`, `photographer_heading`, and `photographer_pitch` columns from the `label` table to the `gsv_data` table.
2. Turn the `label_severity` and `label_temporariness` tables into nullable columns of the `label` table named `severity` and `temporary`.
3. Consider turning the `label_description` table into a nullable column of the `label` table named `description`.

**Comment by misaugstad** [comment]
Adding a bit more onto this. It turns out that the `photographer_heading`, `center_heading`, `origin_heading`, and `pano_yaw_deg` values are all the same number. So we can remove the `center_heading` and `origin_heading` columns from the `gsv_data` table.

**Comment by jonfroehlich** [comment]
Are we sure about this? Was this a mistake on our end in setting up a database schema and then we ended up calling the same function for all of these values... or does GSV actually have those four different things: `photographer_heading`, `center_heading`, `origin_heading`, and `pano_yaw_deg` values? If so, maybe we should keep storing them even if they seem to be the same?

**Comment by misaugstad** [comment]
Finally got a chance to test this a bit more.
* Looks like the `photographer_heading` can sometimes be slightly different from the other values.
* But the `center_heading` and `origin_heading` are identical. And those two are values that we just recently started recording anyway. We have about 100k panos where those values are confirmed identical, and no examples where they are different, so I would feel comfortable removing one of those.
* We were never storing `pano_yaw_deg`, because it isn't available from the JavaScript API. It only shows up in the unofficial XML. And in that XML, it does not include `origin_heading` or `center_heading`, though `pano_yaw_deg` seems to be identical from looking at a few examples. Nothing to change here since we didn't have a way to record it anyway.

**Comment by misaugstad** [comment]
Okay, it looks like `photographer_heading` is in fact identical to `center_heading` and `origin_heading`... What I didn't realize is that the `photographer_heading` value is actually _changing_ occasionally! @ThatOneGoat and I are going to investigate tomorrow.

Similarly, `photographer_pitch` is just the negative of `origin_pitch`, so `photographer_pitch + origin_pitch = 0` everywhere. The differences I was seeing were again from the underlying values changing.

So the current status is that `center_heading`, `origin_heading`, and `origin_pitch`  can all be removed (oops, those are the ones that I just added :grimacing:). Then tomorrow we will investigate why the underlying values are changing. Which will determine if these values should be recorded for every label, or just recorded in the `gsv_data` table and updated whenever the values change.

**Comment by misaugstad** [comment]
Another thing we are seeing is that the `photographer_heading` and `photographer_pitch` will sometimes change by a small margin, typically around 1 degree, rarely more than 3 or 4 degrees (not really sure why). I did not think that this was a value that could change, but it looks like we _did_ actually understand this, judging by the comment and following code [here](https://github.com/ProjectSidewalk/SidewalkWebpage/blob/develop/public/javascripts/SVLabel/src/SVLabel/canvas/Canvas.js#L675).

As of right now, we aren't re-rendering labels that were placed in the past on a pano, but that code would be useful once we merge our open PR that does that ;) And we should try adding that kind of code to the label viewers in LabelMap, Gallery, and the Admin page to see if that improves label placement.

Given that the `photographer_heading` and `photographer_pitch` values are used to compute `sv_image_x` and `sv_image_y`, it would make sense that our cropper would perform better when using the heading/pitch recorded at the time that the label was placed. And after @ThatOneGoat and I did some preliminary testing, that seems to be the case.

Cross-posting with #2485

**Comment by misaugstad** [comment]
Here's a final checklist of changes to make:
- [ ] Turn the `label_severity` table into a nullable column of the `label` table named `severity`
- [ ] Turn the `label_temporariness` table into a nullable column of the `label` table named `temporary`
- [ ] Turn the `label_description` table into a nullable column of the `label` table named `description`
- [ ] Remove `center_heading`, `origin_heading` and `origin_pitch` from the `gsv_data` table
- [ ] Try out new way of rendering old labels on panos in Gallery/LabelMap/Admin

**Comment by misaugstad** [comment]
>Try out new way of rendering old labels on panos in Gallery/LabelMap/Admin

I didn't get anywhere with this. Tried modifying some of the heading values and wasn't really seeing any changes :shrug: Worth a shot!
